Gems/ScriptCanvas/Code/Editor/Utilities/ViewportDragDropHandler.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ViewportDragDropHandler.h"
+
+
+#include <QUrl>
+#include <QDirIterator>
+#include <QMessageBox>
+
+#include <API/EntityCompositionRequestBus.h>
+#include <API/ToolsApplicationAPI.h>
+
+#include <AzCore/Asset/AssetTypeInfoBus.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+
+#include <AzQtComponents/DragAndDrop/ViewportDragAndDrop.h>
+
+#include <Entity/EditorEntityHelpers.h>
+
+#include <ScriptCanvas/Components/EditorScriptCanvasComponent.h>
+
+bool ScriptCanvasAssetDragDropHandler::m_dragAccepted;
+
+namespace PathUtil
+{
+    // Utility to get the game asset folder
+    AZStd::string GetGameAssetsFolder()
```

Comment:
These methods already exists in the EditorCore module (also found a duplicate in ScriptCanvas.Editor.Static for GetEditingGameDataFolder), if its not possible to use them because of dependency inversion, then maybe add a TODO comment to state   that these should be moved to a common module to prevent duplication